packages/xo-web/src/common/xo/index.js

```diff
@@ -2105,6 +2105,38 @@ export const editSr = (sr, { nameDescription, nameLabel }) =>
 export const rescanSr = sr => _call('sr.scan', { id: resolveId(sr) })
 export const rescanSrs = srs => Promise.all(map(resolveIds(srs), id => _call('sr.scan', { id })))
 
+export const toggleSrMaintenanceMode = sr => {
+  const id = resolveId(sr)
+  const method = sr.in_maintenance_mode ? 'disableMaintenanceMode' : 'enableMaintenanceMode'
+
+  return _call(`sr.${method}`, { id }).catch(async err => {
```

Comment:
So we don't ask for confirmation by default?

Reply:
If there are no attached VMs running and there is no side effect of putting the `SR` in maintenance mode, I don't think it is necessary to ask a confirmation.
@julien-f putting the `SR` in maintenance mode can have side effects that the user should be aware of?

Reply:
Maintenance mode will disconnect the SR from all hosts, this preventing it from being used.

It's easily reversible, and I don't think it can have problematic consequences.

/ping @olivierlambert

Reply:
If there's no running VM there's no impact, and since you can reverse the operation without any consequence, I think it's fine to do it that way.